syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.examples.iotservice";
option java_outer_classname = "IoTServiceProto";
option objc_class_prefix = "TEMP";

package iot_service;

// The acceleration service definition.
service IoTService {
  // Responds with a acceleration measurement
  rpc SayAcceleration (AccelerationRequest) returns (AccelerationReply) {}

  rpc CreateUser (UserRequest) returns (StatusReply) {}

  rpc UserLogin (LoginRequest) returns (LoginReply) {}

  // Responds with the current reading of a given light sensor
  rpc SayLightLevel (LightLevelRequest) returns (LightLevelReply) {}
}
message UserRequest {
  string login = 1;
  string password = 2;
  uint32 access = 3;
  uint32 key = 4;
}
message StatusReply {
  string status = 1;
}
message LoginRequest {
  string login = 1;
  string password = 2;
}
message LoginReply {
  string status = 1;
  uint32 access = 2;
  uint32 key = 3;
}

// The request message containing the user's name.
message AccelerationRequest {
  uint32 key = 1;
}

// The response message containing the greetings
message AccelerationReply {
  string status = 1;
  string acceleration = 2;
}

message LightLevelRequest {
  uint32 key = 1;
}

message LightLevelReply {
  string status = 1;
  string lightLevel = 2;
}
